datafusion/core/src/physical_plan/planner.rs

```diff
@@ -1346,6 +1616,74 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_create_cube_expr() -> Result<()> {
+        let logical_plan = test_csv_scan()
+            .await?
+            .project(vec![col("c1"), col("c2"), col("c3")])?
+            .aggregate(
+                vec![cube(vec![col("c1"), col("c2"), col("c3")])],
+                vec![sum(col("c2"))],
+            )?
```

Comment:
I don't understand the need to creating the `aggregate` on the logical plan (as then new cube expressions are planned below). Can you simply use the output of the `project` plan?

The same question applies to the other plans below